### PR TITLE
POST,GET&PUT request works with `is_available` functionality

### PR DESCRIPTION
**Pull request by AzizulHakim96**
### Associated Issues:
- User Stories: #29 
- Dev Tasks: #133 

### Description

This PR does:
1. Changes in the INSERT, UPDATE, and SELECT statement for the new field called `is_available` (`dao\listing.js` file).
2. Change in `dto\listing.js` for adding is_available `required()`.
3. Change in the `rentr_listing` table.

Note:
- COALESCE is not used in the SELECT statement (getAllListing) since it will return the 'not available' apartments too.

### Main References

N/A

### Output Screenshot

- Initial Database:
initalDB

#### GET request test:
- Getting all the listings that are available (should return 2 out of 3):
avaialbilityCheck1

- GetAlllistings work with filters:
availabilityCheck2

#### PUT request test:
- Updating `is_available` works:
update works

-  Getting all the listings that are available (3 out of 3 will return):
getAllListingTest

- Updating `is_available` false again (works):
AgainUpdateWithFalse

- Getting all the listings that are available (2 out of 3 will return):
getAllReturns2

#### POST request test:
- Post with availability success:
PostSuccess

- Getting all the available listings (should return 3 out of 4):
Getting3AvailableListings

### CODE REVIEW

**Review by Mik3Qiao — APPROVED** [review]

File: server/dao/listing.js [hunk 1]
```diff
@@ -77,6 +79,11 @@ class ListingDao {
     const { rows } = await this.dbPool.query(
       `UPDATE rentr_listing
       SET 
+        is_available = (SELECT
+                CASE WHEN is_available = $11 IS NOT NULL THEN $11
```
Comment (NathanCarriere) [inline server/dao/listing.js:83]: Do you think case 11 should go below case 10, just to keep consistent ordering?

File: server/dao/listing.js [hunk 2]
```diff
@@ -8,9 +8,10 @@ class ListingDao {
     const {
       rows,
     } = await this.dbPool.query(
-      'INSERT INTO rentr_listing (userid, title, price, num_bedroom, num_bathroom, is_laundry_available, is_pet_allowed, is_parking_available, images, description) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10) RETURNING id;',
+      'INSERT INTO rentr_listing (userid, is_available, title, price, num_bedroom, num_bathroom, is_laundry_available, is_pet_allowed, is_parking_available, images, description) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11) RETURNING id;',
```
Comment (NathanCarriere) [inline server/dao/listing.js:11]: Does `$11` in this line now refer to `description` and `$2` to `is_available`?

Comment (AzizulHakim96) [reply server/dao/listing.js:11]: Yes, you are correct. May be I can change other line to from 1 to 11 just for the consistency.

File: server/dao/listing.js (hunk 1 above)
Comment (AzizulHakim96) [reply server/dao/listing.js:83]: Yes, I can change the order for consistency.

**Review by jcsnorlax97 — COMMENTED** [review]
Have quickly reviewed it. Looks good to me in general (Good Job, Aziz!). Will review again after you made the changes Nathan mentioned above.

**Review by NathanCarriere — APPROVED** [review]

**Pull request merged**